backend/app/treasury/adapters/covalent/portfolio_v2.py

```diff
@@ -62,7 +61,14 @@ async def get_treasury(
             )
             raise
 
-        db.hset(CACHE_HASH_TREASURY, cache_key, json.dumps(portfolio_data))
+        db.set(cache_key, json.dumps(portfolio_data))
+        if db.ttl(cache_key) <= 0:
+            time_to_evict = datetime.now(tz=dateutil.tz.UTC) + timedelta(days=1)
+            time_to_evict_ts = time_to_evict.replace(
+                hour=0, minute=0, second=0, microsecond=0
+            ).timestamp()
```

Comment:
This logic, namely calculating the eviction time, is used by several pieces of code in the exact same way. That's a good opportunity for factoring it into a helper function, which can be tested.